packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -0,0 +1,248 @@
+package io.flutter.plugins.imagepicker;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.media.MediaScannerConnection;
+import android.net.Uri;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.support.v4.app.ActivityCompat;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.PluginRegistry;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+
+public class ImagePickerDelegate
+    implements PluginRegistry.ActivityResultListener,
+        PluginRegistry.RequestPermissionsResultListener {
+  private static final int REQUEST_CODE_CHOOSE_FROM_GALLERY = 2342;
+  private static final int REQUEST_CODE_TAKE_WITH_CAMERA = 2343;
+  private static final int REQUEST_IMAGE_PICKER_PERMISSIONS = 2344;
+
+  private final Activity registrar;
+  private final ImageResizer imageResizer;
+  private final String providerName;
+
+  private Uri pendingCameraImageUri;
+  private MethodChannel.Result pendingResult;
+  private MethodCall methodCall;
+  private ImagePickingOperationType pendingPickImageOperation;
+
+  public static class FileProvider extends android.support.v4.content.FileProvider {}
```

Comment:
Why this subclass?

Reply:
For this snippet, I actually went through the third-party library internals and saw they used a similar approach. Don't know why I didn't notice that this was actually redundant.